app/src/main/java/com/eleks/mowid/ui/feature/bottomsheet/BottomSheet.kt

```diff
@@ -0,0 +1,128 @@
+import com.eleks.mowid.R
+import com.eleks.mowid.ui.theme.MoWidTheme
+
+@Composable
+fun BottomSheet(
+    bottomSheetUIState: BottomSheetUIState,
+    onButtonClick: (id: String?, text1: String, text2: String) -> Unit,
+    clearSavedStates: Boolean = false
+) {
+
+
+    with(bottomSheetUIState) {
+        var textState1 by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+            mutableStateOf(TextFieldValue(text1))
+        }
+        var textState2 by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+            mutableStateOf(TextFieldValue(text2))
+        }
+
+        if (clearSavedStates) {
+            textState1 = TextFieldValue(text1)
+            textState2 = TextFieldValue(text2)
+            LocalFocusManager.current.clearFocus()
+        }
+
+        Column(
+            modifier = Modifier
+                .fillMaxWidth(),
+            horizontalAlignment = CenterHorizontally
+        ) {
+            Box(
+                modifier = Modifier
+                    .padding(vertical = 16.dp)
+                    .size(width = 32.dp, height = 4.dp)
+                    .clip(CircleShape)
+                    .background(MaterialTheme.colorScheme.outlineVariant)
+            )
+
+            Text(
+                text = stringResource(header),
+                fontSize = 28.sp,
+                style = MaterialTheme.typography.headlineMedium
+            )
+
+            TextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                value = textState1,
+                onValueChange = { textState1 = it },
+                label = {
+                    Text(
+                        text = stringResource(hint1),
+                        style = MaterialTheme.typography.bodySmall
+                    )
+                },
+                singleLine = true,
+                textStyle = MaterialTheme.typography.bodyLarge,
+                keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next)
+            )
+            TextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                value = textState2,
+                onValueChange = { textState2 = it },
+                label = {
+                    Text(
+                        text = stringResource(hint2),
+                        style = MaterialTheme.typography.bodySmall
+                    )
+                },
+                textStyle = MaterialTheme.typography.bodyLarge,
+                keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done)
+            )
+
+            Button(
+                modifier = Modifier.padding(16.dp),
+                enabled = textState1.text.isNotEmpty() && (textState2.text.isNotEmpty()
+                        || (bottomSheetUIState is BottomSheetUIState.AddQuoteBottomSheet || bottomSheetUIState is BottomSheetUIState.EditQuoteBottomSheet)),
```

Comment:
Why do you rely on specific classes of the Bottom Sheet?

Reply:
I would suggest to use property isButtonEnabled in your BottomSheetUIState. This way you can have it in one place and make unit-testable